JS/Lesson18/onur/recursion.js

```diff
@@ -101,18 +133,87 @@ const nestedObj = {
 		linkedIn: 'johndoe',
 	},
 };
+
+function flattenObject(obj, prefix = "") {
+	let resultObject = {};
+	for (const key in obj) {
+		if (typeof obj[key] === 'object' && obj[key] !== null) {
+			const nestedKeys = flattenObject(obj[key], prefix + key + '.');
+			resultObject = { ...resultObject, ...nestedKeys };
+		} else {
+			resultObject[prefix + key] = obj[key];
+		}
+	}
+	return resultObject;
+}
+console.log("Exercise 3:", flattenObject(nestedObj));
+
+console.log('----------------------------');
 // 4. Write a recursive function to count the number of vowels in a string.
 const sentenceEx4 = 'Hello, how are you?';
 const vowels = ['a', 'e', 'i', 'o', 'u'];
 
+function toCountVowelsOfString(sentence, vowels) {
+	if (sentence.length === 0) {
+		return 0;
+	}
+
+	const firstLetter = sentence[0].toLowerCase();
+	if (vowels.includes(firstLetter)) {
+		return 1 + toCountVowelsOfString(sentence.slice(1), vowels);
+	} else {
+		return toCountVowelsOfString(sentence.slice(1), vowels);
+	}
+
+}
+const countVowels = toCountVowelsOfString(sentenceEx4, vowels);
+console.log("Exercise 4: ", countVowels);
+
+console.log('----------------------------');
 // 5. Write a recursive function to remove all occurrences of a specified character from a string.
 const sentenceEx5 = 'Hello, how are you?';
 const charToRemove = 'o';
 
+function toRemoveSpecifiedCharacter(sentence, specifiedCharacter) {
+	if (sentence.length === 0) {
+		return "";
+	}
+	let firstCharacter = sentence[0].toLowerCase();
+	let remainingCharacters = sentence.slice(1);
+	if (firstCharacter === specifiedCharacter) {
+		return toRemoveSpecifiedCharacter(remainingCharacters, specifiedCharacter);
+	} else {
+		return firstCharacter + toRemoveSpecifiedCharacter(remainingCharacters, specifiedCharacter);
+	}
+
+}
+
+const removeCharacter = toRemoveSpecifiedCharacter(sentenceEx5, charToRemove);
+console.log("Exercise 5: ", removeCharacter);
+
+console.log('----------------------------');
 // 6. Write a recursive function to check if an array includes a specific value.
 const numbers = [1, 2, 3, 4, 5];
 const valueToCheck = 3;
 
+function checkToSpesificValue(arr, value, index = 0) {
```

Comment:
This function is hard to name :D It took me at least 5 minutes to come with a name that I'm not completely satisfied with. I'd name it something like 'checkIfValueIncluded'